iscript/src/iscript/mac.py

```diff
@@ -954,11 +955,39 @@ async def create_pkg_files(config, sign_config, all_paths):
                             sign_config["signing_keychain"],
                             "--install-location",
                             "/Applications",
+                        ]
+                        + pkg_opts
+                        + [
                             "--component",
                             app.app_path,
-                            app.pkg_path,
+                            app.tmp_pkg_path,
+                        ],
+                        cwd=app.parent_dir,
+                        exception=IScriptError,
+                    ),
+                )
+            )
+        )
+    await raise_future_exceptions(futures)
+    futures = []
+    for app in all_paths:
+        # Bug 1689376 - distribution pkg
+        futures.append(
+            asyncio.ensure_future(
+                semaphore_wrapper(
+                    semaphore,
+                    run_command(
+                        [
+                            "productbuild",
+                            "--keychain",
+                            sign_config["signing_keychain"],
                         ]
-                        + pkg_opts,
+                        + pkg_opts
+                        + [
+                            "--package",
+                            app.tmp_pkg_path,
+                            app.pkg_path,
+                        ],
```

Comment:
If this is the final pkg, I think we need to run `productsign`, too, otherwise the pkg is totally unsigned. (See https://hg.mozilla.org/build/braindump/file/tip/signing-related/sign-mozilla-vpn.sh#l106 for how we do this for MozillaVPN.)

Reply:
Ah. Maybe that's why mkaply said the pkg installed fine but crashed on startup with a signing error.